core/bin/external_node/src/main.rs

```diff
@@ -424,19 +430,41 @@ async fn run_api(
             )
         });
 
+        let tokens_whitelisted_for_paymaster_cache = Arc::new(RwLock::new(Vec::new()));
+        let tokens_whitelisted_for_paymaster_cache_clone =
+            tokens_whitelisted_for_paymaster_cache.clone();
+        let whitelisted_tokens_update_task = task::spawn(async move {
+            loop {
+                match main_node_client.tokens_whitelisted_for_paymaster().await {
+                    Ok(tokens) => {
+                        *tokens_whitelisted_for_paymaster_cache_clone.write().await = tokens;
+                    }
+                    Err(err) => {
+                        tracing::error!(
+                            "Failed to query `tokens_whitelisted_for_paymaster`, error: {err:?}"
+                        );
```

Comment:
What would be the impact if the whitelisted tokens are desynced between the main node and an EN? Would the VM-related RPC methods work incorrectly? (I suppose we cannot avoid temporary desyncing in any case, unless we sync via an L2 smart contract or something like that.)

Reply:
If there are tokens that are in main node's list but not on EN, then EN will reject all transactions that call this token addresses during validation, the most popular scenario is when paymaster calls token contract.
If there are tokens that are in EN's list but not on main node, then EN can accept some transactions that will be rejected by main node.